elastic/kibana
# [Fleet] Do not allow to add Fleet servers in serverless

## Summary

We should not be able to create a Fleet server policy in serverless as it's not needed when Fleet server is running in standalone.

That PR does that by:
* hidding the Add Fleet Server button in the UI
* filtering the Fleet server integration, in the search and while trying to install it.


### UI Changes

##### Installing Fleet server fail

<img width="1504" alt="Screenshot 2023-06-22 at 11 12 08 AM" src="https://github.com/elastic/kibana/assets/1336873/6ddda31f-7684-4ba9-a2af-f4d591fe2f1c">


##### Integrations list fleet server is not present

<img width="1344" alt="Screenshot 2023-06-22 at 11 11 28 AM" src="https://github.com/elastic/kibana/assets/1336873/5528efa5-c0be-4a3e-99c4-18b1b2ee986d">


##### Remove add fleet server host buttons in settings 
<img width="1338" alt="Screenshot 2023-06-21 at 1 41 36 PM" src="https://github.com/elastic/kibana/assets/1336873/6a423ff0-6966-4032-86c4-4f0a53b4d992">

##### Remove add fleet server host from the agent search bar
<img width="1317" alt="Screenshot 2023-06-21 at 1 48 21 PM" src="https://github.com/elastic/kibana/assets/1336873/c5f0ddf4-5f9a-44de-a917-81a9df6fdd70">


### Manual test instructions

Add the following config to simulate you are in serverless and check that you cannot create a Fleet server policy.

```
xpack.fleet.internal.fleetServerStandalone: true
xpack.fleet.fleetServerHosts:
  - id: test
    is_default: true
    name: Test
    host_urls: ['https://fleetserver:8220']
```

### Todo
